src/com/strings/LongestCommonPrefix.java

```diff
@@ -0,0 +1,29 @@
+package com.strings;
+
+public class LongestCommonPrefix {
+    public static String longestCommonPrefix(String[] strs) {
+        if (strs == null || strs.length == 0) {
+            return ""; // to not get error on empty string
+        }
+
+        String smallest = strs[0]; //to get a reference for comparing
+
+        System.out.println(smallest);
+        for (String str : strs) {
+            while (!str.startsWith(smallest)) {
```

Comment:
try to solve problem with indexOf, it seems you will get better performance with it